src/components/scene/background.js

```diff
@@ -1,49 +1,222 @@
+
+    this.setupLightsForLightingEstimation();
+
+    this.glBinding = new XRWebGLBinding(xrSession, gl);
+    gl.getExtension('EXT_sRGB');
+    gl.getExtension('OES_texture_half_float');
+
+    xrSession.requestLightProbe()
+      .then(function (lightProbe) {
+        scene.environment = self.lightProbeTarget.texture;
+
+        self.xrLightProbe = lightProbe;
+        lightProbe.addEventListener('reflectionchange', function onReflectionChanged () {
+          self.needsEnvironmentUpdate = true;
+        });
+      })
+      .catch(function (err) {
+        console.warn('Lighting estimation not supported: ' + err.message);
+        console.warn('Are you missing: webxr="optionalFeatures: light-estimation;" from <a-scene>?');
+      });
   },
   update: function () {
     var scene = this.el.sceneEl.object3D;
     var data = this.data;
     var object3D = this.el.object3D;
+
     if (data.transparent) {
       object3D.background = null;
     } else {
       object3D.background = new THREE.Color(data.color);
     }
 
-    if (scene.environment && scene.environment !== this.cubeRenderTarget.texture) {
-      console.warn('Background will not overide predefined environment maps');
+    if (scene.environment &&
+      (scene.environment !== this.cubeRenderTarget.texture || scene.environment !== this.lightProbeTarget.texture)
+    ) {
+      console.warn('Background will not override user defined environment maps');
       return;
     }
 
     if (data.generateEnvironment) {
       scene.environment = this.cubeRenderTarget.texture;
+      this.needsEnvironmentUpdate = true;
     } else {
       scene.environment = null;
     }
   },
 
-  tick: function () {
-    if (!this.needsEnvironmentUpdate) return;
+  updateXRCubeMap: function () {
+    // Update Cube Map
+    var renderer = this.el.renderer;
+    var cubeMap = this.glBinding.getReflectionCubeMap(this.xrLightProbe);
+    if (cubeMap) {
+      var rendererProps = renderer.properties.get(this.lightProbeTarget.texture);
+      rendererProps.__webglTexture = cubeMap;
+    } else {
+      console.log('Cube map not available');
+    }
+  },
+
+  tick: function (time, delta) {
     var scene = this.el.object3D;
     var renderer = this.el.renderer;
-    var camera = this.el.camera;
+    var frame = this.el.sceneEl.frame;
+
+    this.timeSinceUpdate += delta;
+    if (
+      this.data.environmentUpdateFrequency > 0 && // should update in general?
+      this.timeSinceUpdate > (this.data.environmentUpdateFrequency * 1000) // should update this tick?
+    ) {
+      this.timeSinceUpdate = 0;
+      this.needsEnvironmentUpdate = true;
+    }
+
+    if (frame && this.xrLightProbe) {
+      var estimate = frame.getLightEstimate(this.xrLightProbe);
+
+      if (estimate) {
+        updateLights(
+          estimate,
+          this.probeLight.components.light.light,
+          this.directionalLight.components.light.light
+        );
+      } else {
+        console.log('light estimate not yet available');
```

Comment:
That's because the browser doesn't support the API or because an estimate has not yet been calculated?

Reply:
It hasn't yet been calculated, I find I normally get this message for 6 frames and then it's working as expected.

Reply:
I see. it looks that estimation has some sort of calibration / bootstrap process? Is it worth to display a message or is it noisy? Maybe a note for the developer would suffice? 

```
 // Light estimation might not be available: initialization, calculation time...
  if (estimate) {
         updateLights(
           estimate,
           this.probeLight.components.light.light,
           this.directionalLight.components.light.light
         );
       } 
```

Reply:
Sure. No point adding unneeded console noise.